### DISCUSSION TITLE
Temporary coerce

### DISCUSSION THREAD

**Discussion by olivermaldonado** [open-circle/valibot#455] [Q&A] [answered]
Is there a way to temporarily transform or coerce an input, run it through some verifications, and return the original input?

For example, create a schema that accepts a number, that passes some number-specific validations, but that can also accept a string representation of the number, and pass it through the same validations, coerced as a number. Then output the original input value, in this example either a string or a number.

Right now the coerce function will irrevocably change the type.

**Comment by fabian-hiller** [comment, marked as answer]
Something like that could work. Check it out in our [playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgGUmALAUxDI4AXkwA6AK6tgbABT04cANoYxHKMFYBzWSrFQ+WvgA9ZAegB6AWgD8AHQAmAalk2AXHbGOnAShsASMx8AXR8AGnFWCRAKPihZELCFZWTFVSYJDnxZWU0wCRgfEQA+OAA5aNj4vIKi4tEABgjCCpi4uGB0GAgIOFRBUlJCcNTxDKyQHJrCkvLKuNzWfJmAHlEARgamolaqjq6euApgLWGkxWD6HwZmNlQIUj4xUggdVVRkbD4ABWQoVD4sm4-EEEQAROsAExgnxFMxmOAQ6G3dgPJ4vN5qT4-P4AoG8ATICJQ2FweFwKGMFiox5iOJ4eLvbG-f6A4GE8FWdYwuEIgCiUAZKPutPp0FkTK+LLx7NBcC5pPJAoZQA).

```ts
const Schema = v.union(
  [v.string([v.regex(/^-?\d+(?:\.\d+)?$/)]), v.number()],
  [
    v.custom((input) => Number(input) >= 0, 'Number is too small'),
    v.custom((input) => Number(input) <= 100, 'Number is too big'),
  ]
);
```

**Reply by olivermaldonado** [reply]
Yes. Using custom validators will work. I was trying to use the predefined validators included in the library.

I was trying to make a function that returns a predefined schema that would accept an array of unknown validators to check against a number type as a last step, but that would also accept (and return) string representations.

By only using custom validators it will work. Thanks.
